### ISSUE TITLE
[bug] `SyntaxWarning` spam from `patch_ng` on Conan start with Python 3.12

### ISSUE THREAD

**Issue by Artalus** [conan-io/conan#16759]
### Describe the bug

Starting with Python 3.12, Conan verbosely complains about deprecated syntax in `patch_ng`:
```
$ conan
/tmp/conan2/.venv/lib/python3.12/site-packages/patch_ng.py:427: SyntaxWarning: invalid escape sequence '\d'
  re_hunk_start = re.compile(b"^@@ -(\d+)(,(\d+))? \+(\d+)(,(\d+))? @@")
/tmp/conan2/.venv/lib/python3.12/site-packages/patch_ng.py:578: SyntaxWarning: invalid escape sequence '\s'
  re_filename_date_time = b"^--- ([^\t]+)(?:\s([0-9-]+)\s([0-9:]+)|.*)"
/tmp/conan2/.venv/lib/python3.12/site-packages/patch_ng.py:621: SyntaxWarning: invalid escape sequence '\+'
  re_filename_date_time = b"^\+\+\+ ([^\t]+)(?:\s([0-9-]+)\s([0-9:]+)|.*)"
/tmp/conan2/.venv/lib/python3.12/site-packages/patch_ng.py:653: SyntaxWarning: invalid escape sequence '\d'
  match = re.match(b"^@@ -(\d+)(,(\d+))? \+(\d+)(,(\d+))? @@(.*)", line)

Consumer commands
cache      Perform file operations in the local cache (of recipes and/or packages).
...
```
What was `DeprecationWarning` in older Pythons is now a `SyntaxWarning` - which is probably why they now appear in Conan output.

Could likely be fixed by merging conan-io/python-patch-ng#29 , releasing new version and bumping the dep in Conan.

### How to reproduce it

- Install py 3.12
- Install conan
- Run any command

**Comment by uilianries** [comment]
The PR https://github.com/conan-io/python-patch-ng/pull/29 has been merged (kudos Artalus). We should provide a new release of python-patch, then, align it in conan client. Thank you for reporting the issue!

**Comment by uilianries** [comment]
The new release https://pypi.org/project/patch-ng/1.18.0/ has a proper fix available. Need to update Conan client to install the latest version.

**Comment by uilianries** [comment]
Hello @Artalus I can not reproduce your error. Could you please give more details:

- Conan Version
- OS
- Python version (including patch version)
- How did you install Conan? (pip, brew, ...)
- How did you install Python? (apt, pyenv, ...)

No Warnings using Conan 2.6.0 (latest). The `patch-ng` version is 1.17.4:

```
docker run --rm python:3.12.4-slim /bin/bash -c "pip install conan==2.6.0 && conan"
Collecting conan==2.6.0
  Downloading conan-2.6.0.tar.gz (449 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 449.9/449.9 kB 1.4 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting requests<3.0.0,>=2.25 (from conan==2.6.0)
  Downloading requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Collecting urllib3<1.27,>=1.26.6 (from conan==2.6.0)
  Downloading urllib3-1.26.19-py2.py3-none-any.whl.metadata (49 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.3/49.3 kB 2.3 MB/s eta 0:00:00
Collecting colorama<0.5.0,>=0.4.3 (from conan==2.6.0)
  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting PyYAML<7.0,>=6.0 (from conan==2.6.0)
  Downloading PyYAML-6.0.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)
Collecting patch-ng<1.18,>=1.17.4 (from conan==2.6.0)
  Downloading patch-ng-1.17.4.tar.gz (17 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting fasteners>=0.15 (from conan==2.6.0)
  Downloading fasteners-0.19-py3-none-any.whl.metadata (4.9 kB)
Collecting Jinja2<4.0.0,>=3.0 (from conan==2.6.0)
  Downloading jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)
Collecting python-dateutil<3,>=2.8.0 (from conan==2.6.0)
  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting distro<=1.8.0,>=1.4.0 (from conan==2.6.0)
  Downloading distro-1.8.0-py3-none-any.whl.metadata (6.9 kB)
Collecting MarkupSafe>=2.0 (from Jinja2<4.0.0,>=3.0->conan==2.6.0)
  Downloading MarkupSafe-2.1.5-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.0 kB)
Collecting six>=1.5 (from python-dateutil<3,>=2.8.0->conan==2.6.0)
  Downloading six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Collecting charset-normalizer<4,>=2 (from requests<3.0.0,>=2.25->conan==2.6.0)
  Downloading charset_normalizer-3.3.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (33 kB)
Collecting idna<4,>=2.5 (from requests<3.0.0,>=2.25->conan==2.6.0)
  Downloading idna-3.7-py3-none-any.whl.metadata (9.9 kB)
Collecting certifi>=2017.4.17 (from requests<3.0.0,>=2.25->conan==2.6.0)
  Downloading certifi-2024.7.4-py3-none-any.whl.metadata (2.2 kB)
Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Downloading distro-1.8.0-py3-none-any.whl (20 kB)
Downloading fasteners-0.19-py3-none-any.whl (18 kB)
Downloading jinja2-3.1.4-py3-none-any.whl (133 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.3/133.3 kB 4.3 MB/s eta 0:00:00
Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 229.9/229.9 kB 4.1 MB/s eta 0:00:00
Downloading PyYAML-6.0.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (724 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 725.0/725.0 kB 5.3 MB/s eta 0:00:00
Downloading requests-2.32.3-py3-none-any.whl (64 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.9/64.9 kB 5.1 MB/s eta 0:00:00
Downloading urllib3-1.26.19-py2.py3-none-any.whl (143 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.9/143.9 kB 4.2 MB/s eta 0:00:00
Downloading certifi-2024.7.4-py3-none-any.whl (162 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 163.0/163.0 kB 3.7 MB/s eta 0:00:00
Downloading charset_normalizer-3.3.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (141 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 141.9/141.9 kB 3.7 MB/s eta 0:00:00
Downloading idna-3.7-py3-none-any.whl (66 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 66.8/66.8 kB 2.9 MB/s eta 0:00:00
Downloading MarkupSafe-2.1.5-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (28 kB)
Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Building wheels for collected packages: conan, patch-ng
  Building wheel for conan (pyproject.toml): started
  Building wheel for conan (pyproject.toml): finished with status 'done'
  Created wheel for conan: filename=conan-2.6.0-py3-none-any.whl size=572728 sha256=41050e021108873af6cb61cf6a4605a6f13ee435ed6190cbca9e5fb2dd358e54
  Stored in directory: /root/.cache/pip/wheels/c6/c5/be/6bb2224a74c2abbe1b10c25dafa5115374fe2a2e28f95362cd
  Building wheel for patch-ng (setup.py): started
  Building wheel for patch-ng (setup.py): finished with status 'done'
  Created wheel for patch-ng: filename=patch_ng-1.17.4-py3-none-any.whl size=16075 sha256=951ce66c81ac2ed66e5974983f37e8eba7c010bf7072c3f068a9bb491403eac5
  Stored in directory: /root/.cache/pip/wheels/c0/ca/79/d6c49e498d31e473f26449c5aaa3ffdeb4d9d74b48407f8067
Successfully built conan patch-ng
Installing collected packages: patch-ng, urllib3, six, PyYAML, MarkupSafe, idna, fasteners, distro, colorama, charset-normalizer, certifi, requests, python-dateutil, Jinja2, conan
Successfully installed Jinja2-3.1.4 MarkupSafe-2.1.5 PyYAML-6.0.1 certifi-2024.7.4 charset-normalizer-3.3.2 colorama-0.4.6 conan-2.6.0 distro-1.8.0 fasteners-0.19 idna-3.7 patch-ng-1.17.4 python-dateutil-2.9.0.post0 requests-2.32.3 six-1.16.0 urllib3-1.26.19
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: pip install --upgrade pip

Consumer commands
cache      Perform file operations in the local cache (of recipes and/or packages).
config     Manage the Conan configuration in the Conan home.
graph      Compute a dependency graph, without installing or building the binaries.
inspect    Inspect a conanfile.py to return its public fields.
install    Install the requirements specified in a recipe (conanfile.py or conanfile.txt).
list       List existing recipes, revisions, or packages in the cache (by default) or the
             remotes.
lock       Create or manage lockfiles.
pkglist    Several operations over package lists
profile    Manage profiles.
remote     Manage the remote list and the users authenticated on them.
remove     Remove recipes or packages from local cache or a remote.
search     Search for package recipes in all the remotes (by default), or a remote.
version    Give information about the Conan client version.

Creator commands
build      Install dependencies and call the build() method.
create     Create a package.
download   Download (without installing) a single conan package from a remote server.
editable   Allow working with a package that resides in user folder.
export     Export a recipe to the Conan package cache.
export-pkg Create a package directly from pre-compiled binaries.
new        Create a new example recipe and source files from a template.
source     Call the source() method.
test       Test a package from a test_package folder.
upload     Upload packages to a remote.

Type "conan <command> -h" for help
```

Same for Conan 1.x:

```
docker run --rm python:3.12.4-slim /bin/bash -c "pip install conan==1.64.1 && conan"
Collecting conan==1.64.1
  Downloading conan-1.64.1.tar.gz (789 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 789.4/789.4 kB 2.6 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting requests<3.0.0,>=2.25 (from conan==1.64.1)
  Downloading requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Collecting urllib3<1.27,>=1.26.6 (from conan==1.64.1)
  Downloading urllib3-1.26.19-py2.py3-none-any.whl.metadata (49 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.3/49.3 kB 5.2 MB/s eta 0:00:00
Collecting colorama<0.5.0,>=0.3.3 (from conan==1.64.1)
  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting PyYAML<6.1,>=3.11 (from conan==1.64.1)
  Downloading PyYAML-6.0.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)
Collecting patch-ng<1.18,>=1.17.4 (from conan==1.64.1)
  Downloading patch-ng-1.17.4.tar.gz (17 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting fasteners>=0.14.1 (from conan==1.64.1)
  Downloading fasteners-0.19-py3-none-any.whl.metadata (4.9 kB)
Collecting six<=1.16.0,>=1.10.0 (from conan==1.64.1)
  Downloading six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Collecting node-semver==0.6.1 (from conan==1.64.1)
  Downloading node_semver-0.6.1-py3-none-any.whl.metadata (2.1 kB)
Collecting distro<=1.8.0,>=1.0.2 (from conan==1.64.1)
  Downloading distro-1.8.0-py3-none-any.whl.metadata (6.9 kB)
Collecting pygments<3.0,>=2.0 (from conan==1.64.1)
  Downloading pygments-2.18.0-py3-none-any.whl.metadata (2.5 kB)
Collecting tqdm<5,>=4.28.1 (from conan==1.64.1)
  Downloading tqdm-4.66.4-py3-none-any.whl.metadata (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.6/57.6 kB 4.3 MB/s eta 0:00:00
Collecting Jinja2<4.0.0,>=3.0 (from conan==1.64.1)
  Downloading jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)
Collecting python-dateutil<3,>=2.7.0 (from conan==1.64.1)
  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting bottle<0.13,>=0.12.8 (from conan==1.64.1)
  Downloading bottle-0.12.25-py3-none-any.whl.metadata (1.8 kB)
Collecting pluginbase>=0.5 (from conan==1.64.1)
  Downloading pluginbase-1.0.1.tar.gz (43 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.6/43.6 kB 4.4 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting PyJWT<3.0.0,>=2.4.0 (from conan==1.64.1)
  Downloading PyJWT-2.9.0-py3-none-any.whl.metadata (3.0 kB)
Collecting MarkupSafe>=2.0 (from Jinja2<4.0.0,>=3.0->conan==1.64.1)
  Downloading MarkupSafe-2.1.5-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.0 kB)
Collecting charset-normalizer<4,>=2 (from requests<3.0.0,>=2.25->conan==1.64.1)
  Downloading charset_normalizer-3.3.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (33 kB)
Collecting idna<4,>=2.5 (from requests<3.0.0,>=2.25->conan==1.64.1)
  Downloading idna-3.7-py3-none-any.whl.metadata (9.9 kB)
Collecting certifi>=2017.4.17 (from requests<3.0.0,>=2.25->conan==1.64.1)
  Downloading certifi-2024.7.4-py3-none-any.whl.metadata (2.2 kB)
Downloading node_semver-0.6.1-py3-none-any.whl (10 kB)
Downloading bottle-0.12.25-py3-none-any.whl (90 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.2/90.2 kB 4.2 MB/s eta 0:00:00
Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Downloading distro-1.8.0-py3-none-any.whl (20 kB)
Downloading fasteners-0.19-py3-none-any.whl (18 kB)
Downloading jinja2-3.1.4-py3-none-any.whl (133 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.3/133.3 kB 5.3 MB/s eta 0:00:00
Downloading pygments-2.18.0-py3-none-any.whl (1.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 3.7 MB/s eta 0:00:00
Downloading PyJWT-2.9.0-py3-none-any.whl (22 kB)
Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 229.9/229.9 kB 4.3 MB/s eta 0:00:00
Downloading PyYAML-6.0.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (724 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 725.0/725.0 kB 5.2 MB/s eta 0:00:00
Downloading requests-2.32.3-py3-none-any.whl (64 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.9/64.9 kB 6.4 MB/s eta 0:00:00
Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Downloading tqdm-4.66.4-py3-none-any.whl (78 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.3/78.3 kB 2.7 MB/s eta 0:00:00
Downloading urllib3-1.26.19-py2.py3-none-any.whl (143 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.9/143.9 kB 4.2 MB/s eta 0:00:00
Downloading certifi-2024.7.4-py3-none-any.whl (162 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 163.0/163.0 kB 336.5 kB/s eta 0:00:00
Downloading charset_normalizer-3.3.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (141 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 141.9/141.9 kB 534.8 kB/s eta 0:00:00
Downloading idna-3.7-py3-none-any.whl (66 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 66.8/66.8 kB 8.1 MB/s eta 0:00:00
Downloading MarkupSafe-2.1.5-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (28 kB)
Building wheels for collected packages: conan, patch-ng, pluginbase
  Building wheel for conan (setup.py): started
  Building wheel for conan (setup.py): finished with status 'done'
  Created wheel for conan: filename=conan-1.64.1-py3-none-any.whl size=923284 sha256=e1af67452b0cd6357037a16c3a436600c56a9d25030d21677d82cee34c664917
  Stored in directory: /root/.cache/pip/wheels/4f/74/bb/738a9ffdebe227a6eb4b7fa307b66aeafb36d44dc6f4ee84cf
  Building wheel for patch-ng (setup.py): started
  Building wheel for patch-ng (setup.py): finished with status 'done'
  Created wheel for patch-ng: filename=patch_ng-1.17.4-py3-none-any.whl size=16075 sha256=8a86aac832adf6c2e1fce20c6c11fba931146295822de2df2608b6ee956bc410
  Stored in directory: /root/.cache/pip/wheels/c0/ca/79/d6c49e498d31e473f26449c5aaa3ffdeb4d9d74b48407f8067
  Building wheel for pluginbase (setup.py): started
  Building wheel for pluginbase (setup.py): finished with status 'done'
  Created wheel for pluginbase: filename=pluginbase-1.0.1-py3-none-any.whl size=7796 sha256=844268279cab8971fbcd6fd7c7e1ede6c4f26989fe3ffe38a9c6939293cd498e
  Stored in directory: /root/.cache/pip/wheels/5d/d2/56/baac190d6d214e0b4d7c763d8a6dd388e0dfb2690283ec18fa
Successfully built conan patch-ng pluginbase
Installing collected packages: pluginbase, patch-ng, node-semver, bottle, urllib3, tqdm, six, PyYAML, PyJWT, pygments, MarkupSafe, idna, fasteners, distro, colorama, charset-normalizer, certifi, requests, python-dateutil, Jinja2, conan
Successfully installed Jinja2-3.1.4 MarkupSafe-2.1.5 PyJWT-2.9.0 PyYAML-6.0.1 bottle-0.12.25 certifi-2024.7.4 charset-normalizer-3.3.2 colorama-0.4.6 conan-1.64.1 distro-1.8.0 fasteners-0.19 idna-3.7 node-semver-0.6.1 patch-ng-1.17.4 pluginbase-1.0.1 pygments-2.18.0 python-dateutil-2.9.0.post0 requests-2.32.3 six-1.16.0 tqdm-4.66.4 urllib3-1.26.19
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: pip install --upgrade pip
Consumer commands
  install    Installs the requirements specified in a recipe (conanfile.py or conanfile.txt).
  config     Manages Conan configuration.
  get        Gets a file or list a directory of a given reference or package.
  info       Gets information about the dependency graph of a recipe.
  search     Searches package recipes and binaries in the local cache or a remote. Unless a
             remote is specified only the local cache is searched.
Creator commands
  new        Creates a new package recipe template with a 'conanfile.py' and optionally,
             'test_package' testing files.
  create     Builds a binary package for a recipe (conanfile.py).
  upload     Uploads a recipe and binary packages to a remote.
  export     Copies the recipe (conanfile.py & associated files) to your local cache.
  export-pkg Exports a recipe, then creates a package from local source and build folders.
  test       Tests a package consuming it from a conanfile.py with a test() method.
Package development commands
  source     Calls your local conanfile.py 'source()' method.
  build      Calls your local conanfile.py 'build()' method.
  package    Calls your local conanfile.py 'package()' method.
  editable   Manages editable packages (packages that reside in the user workspace, but are
             consumed as if they were in the cache).
  workspace  Manages a workspace (a set of packages consumed from the user workspace that
             belongs to the same project).
Misc commands
  profile    Lists profiles in the '.conan/profiles' folder, or shows profile details.
  remote     Manages the remote list and the package recipes associated with a remote.
  user       Authenticates against a remote with user/pass, caching the auth token.
  imports    Calls your local conanfile.py or conanfile.txt 'imports' method.
  copy       Copies conan recipes and packages to another user/channel.
  remove     Removes packages or binaries matching pattern from local cache or remote.
  alias      Creates and exports an 'alias package recipe'.
  download   Downloads recipe and binaries to the local cache, without using settings.
  inspect    Displays conanfile attributes, like name, version, and options. Works locally,
             in local cache and remote.
  help       Shows help for a specific command.
  lock       Generates and manipulates lock files.
  frogarian  Conan The Frogarian

Conan commands. Type "conan <command> -h" for help
```

**Comment by Artalus** [comment]
Huh. I might be having a brain rot - was 100% positive I first encountered this in CI while experimenting with new docker image + new conan, but cannot reproduce it now either...

I got the quote from the starting post by using https://github.com/astral-sh/uv , a `pip` replacement that I got severely addicted to. Simplest way to reproduce is via `uv run`
```
artalus@idpad  /tmp $ uv run --with 'conan>2' conan
warning: `uv run` is experimental and may change without warning.
Resolved 15 packages in 1.24s
Installed 15 packages in 12ms
 + certifi==2024.7.4
 + charset-normalizer==3.3.2
 + colorama==0.4.6
 + conan==2.6.0
 + distro==1.8.0
 + fasteners==0.19
 + idna==3.7
 + jinja2==3.1.4
 + markupsafe==2.1.5
 + patch-ng==1.17.4
 + python-dateutil==2.9.0.post0
 + pyyaml==6.0.1
 + requests==2.32.3
 + six==1.16.0
 + urllib3==1.26.19
/home/artalus/.cache/uv/environments-v0/.tmpLEhMlg/lib/python3.12/site-packages/patch_ng.py:427: SyntaxWarning: invalid escape sequence '\d'
  re_hunk_start = re.compile(b"^@@ -(\d+)(,(\d+))? \+(\d+)(,(\d+))? @@")
/home/artalus/.cache/uv/environments-v0/.tmpLEhMlg/lib/python3.12/site-packages/patch_ng.py:578: SyntaxWarning: invalid escape sequence '\s'
  re_filename_date_time = b"^--- ([^\t]+)(?:\s([0-9-]+)\s([0-9:]+)|.*)"
/home/artalus/.cache/uv/environments-v0/.tmpLEhMlg/lib/python3.12/site-packages/patch_ng.py:621: SyntaxWarning: invalid escape sequence '\+'
  re_filename_date_time = b"^\+\+\+ ([^\t]+)(?:\s([0-9-]+)\s([0-9:]+)|.*)"
/home/artalus/.cache/uv/environments-v0/.tmpLEhMlg/lib/python3.12/site-packages/patch_ng.py:653: SyntaxWarning: invalid escape sequence '\d'
  match = re.match(b"^@@ -(\d+)(,(\d+))? \+(\d+)(,(\d+))? @@(.*)", line)

Consumer commands
cache      Perform file operations in the local cache (of recipes and/or packages).
...
```
, or with `uv venv`:
```
artalus@idpad  /tmp/c $ uv venv
Using Python 3.12.4 interpreter at: /usr/bin/python3
Creating virtualenv at: .venv
Activate with: source .venv/bin/activate

artalus@idpad  /tmp/c $ uv pip install conan
Resolved 15 packages in 1.11s
Installed 15 packages in 11ms
 + certifi==2024.7.4
 + charset-normalizer==3.3.2
 + colorama==0.4.6
 + conan==2.6.0
 + distro==1.8.0
 + fasteners==0.19
 + idna==3.7
 + jinja2==3.1.4
 + markupsafe==2.1.5
 + patch-ng==1.17.4
 + python-dateutil==2.9.0.post0
 + pyyaml==6.0.1
 + requests==2.32.3
 + six==1.16.0
 + urllib3==1.26.19

artalus@idpad  /tmp/c $ source .venv/bin/activate

artalus@idpad  /tmp/c $ conan
/tmp/c/.venv/lib/python3.12/site-packages/patch_ng.py:427: SyntaxWarning: invalid escape sequence '\d'
  re_hunk_start = re.compile(b"^@@ -(\d+)(,(\d+))? \+(\d+)(,(\d+))? @@")
/tmp/c/.venv/lib/python3.12/site-packages/patch_ng.py:578: SyntaxWarning: invalid escape sequence '\s'
  re_filename_date_time = b"^--- ([^\t]+)(?:\s([0-9-]+)\s([0-9:]+)|.*)"
/tmp/c/.venv/lib/python3.12/site-packages/patch_ng.py:621: SyntaxWarning: invalid escape sequence '\+'
  re_filename_date_time = b"^\+\+\+ ([^\t]+)(?:\s([0-9-]+)\s([0-9:]+)|.*)"
/tmp/c/.venv/lib/python3.12/site-packages/patch_ng.py:653: SyntaxWarning: invalid escape sequence '\d'
  match = re.match(b"^@@ -(\d+)(,(\d+))? \+(\d+)(,(\d+))? @@(.*)", line)

Consumer commands
cache      Perform file operations in the local cache (of recipes and/or packages).
```

I also noticed that if you use `uv venv`, the warnings appear only once - subsequent `conan` calls do not produce them...
Did a quick check but did not see anything suspicious, like `PYTHONWARNINGS` envvar set, that could've caused the difference from a regular venv 🤷

**Comment by uilianries** [comment]
@Artalus Okay, I understand your point. In the patch-ng was capable to see the same warning when running those tests in the CI. Indeed there is some configuration suppressing it. Thank you!
